### ISSUE TITLE
Feature suggestion (that may not be a good fit): open folder...?

### ISSUE THREAD

**Issue by ell1e** [orbitalquark/textadept#50]
I really like the basic text adept UI, but there is one major UI feature missing that makes me honestly a bit lost after looking around it for 4-5 minutes how to make it work for me. Now I realize this is just how I like the IDEs, although I think what I am thinking of is one of the most popular paradigms in modern IDEs and how Atom and VS Code for example tend to encourage use, but still - maybe not a good fit for textadept. But I want to suggest it anyway, since I think it'd be nice to have:

My suggestion is basically:

1. Rename `File > Open` to `File > Open File`, and add `File > Open Folder` right below
2. When `File > Open Folder` is clicked and a folder is chosen in a folder picker, a sidebar will appear with a document browser of that folder to quickly open files in tabs

That's already it. But I realize the document browser is a large clutter addition to the UI. However, since it would only show up when the `File > Open Folder` entry is used (I would suggest for it to just not show up otherwise), I feel like at least it wouldn't distract users who prefer the more minimal current UI much. And I just feel like the current paradigm of manually opening files with optional sessions just doesn't work too well with large projects which are without a quick document browser just not too fun to browse, IMHO. Or maybe it's just me? But I think it'd be possibly a great addition, if in your opinion it doesn't strive too far from the otherwise fairly minimal base.

**Comment by Andre-LA** [comment]
There was a module (which maybe I'm the only one that have it now bc the author deleted the account) that you could open a folder and make it as the "project".

I my opinion, this option (or something similar) should be added to textadept indeed, where you can open a directory as the project, instead of making it a exclusive when opening a repository.

This also would be useful even in repositories, for example, in Unity projects there are a lot more files than just code files, using this feature I could only open the Assets directory (which is where the code files would be), and ignore the rest.

**Comment by rgieseke** [comment]
https://github.com/gabdub/ta-tweaks has a project browser.

You could also build something like you imagine with https://github.com/rgieseke/textredux/

**Comment by ell1e** [comment]
Right, but there is a problem with recommending textadept to some users if this doesn't work out of the box. It just can be a slightly confusing initial experience, like "ok but how do I even open my project in this to work on it"? So it's nice that plugins exist, but I'm mostly concerned about the out of the box experience. But this is why I made the it might not be a good fit remarks, maybe textadept just needs to remain so super minimal that this cannot be included by default. However, I think it would be worth a consideration which is why I made the ticket.

Basically, at least me personally when I need to install a plugin just to browse the files I just tend to move on. But again, that might just be me, and maybe I'm just not the target audience. But I feel like one more menu entry + a sidebar that only shows up when that menu entry is used might be worth an inclusion to make textadept more mainstream accessible.

**Comment by rgieseke** [comment]
Good point, actually Textadept once had a sidebar  panel, it was removed after 2.1 (when the filtered list was introduced).
@orbitalquark likely has more thoughts on this.

**Comment by ell1e** [comment]
Maybe some more context of how I'm thinking about this, from a pedagogic/teaching angle:

In a university course or the like, a common approach might be this:

1. Lecture 1-3: work in a single file, learn basics of lua/python/what have you, for which Textadept is a really great match

2. Lecture 4: "ok now an interlude on repo utter basics, and git for windows & clone, so you can fetch a prepared sample project to change something in". (Added in primer on future collaboration tooling, basically.) Please note whether a course covers this at all is obviously a highly subjective thing, since some focus *just* on the programming and might leave this out. But from my experience it is increasingly more common to at least cover what a repo is somewhere in the course, and specifically to prioritize this over the use of a traditional IDE since these seem to be slightly fading out anyway and it's more important to figure out how to share code with whatever editor works for you these days rather than learn all the Visual Studio buttons. However, now with Textadept we'd hit a bit of a ditch: either all the students need to be using university equipment where a plugin can  be easily preinstalled which is often NOT the case (bring your own laptop is very common), or suddenly they all need to install a plugin so they can easily open the sample project up. Of course this is not a complete showstopper and most courses would just have a quick line on how to add in the plugin, but it might be a reason to e.g. just use Atom or something for the course instead which appeals slightly more to a wider range of coders anyway in parts due to this "Open Folder" paradigm.

Basically I'm giving this example to show that even to quite the beginners, this feature **might** be relevant. Specifically also when introducing beginners to a new IDE, this could make textadept a more attractive target since it is otherwise very beginner friendly in ways something like Atom might be less so.

**Comment by rgieseke** [comment]
Here's a screenshot of how it looked in 2.1
![image](https://user-images.githubusercontent.com/198537/106966571-e8bc7000-6745-11eb-9dc5-a6053dd18afb.png)

**Comment by ell1e** [comment]
Oh nice. Yeah that looks pretty much exactly what I would suggest. And I think it would be totally fine to not show this unless "Open Folder" is ever used, so the default UI opening up would still remain super clean.

**Comment by orbitalquark** [comment]
I wrote a simple text-based file browser module that operates in a split view: https://github.com/orbitalquark/textadept/wiki/ta-filebrowser. I just updated it for Textadept 11.

I am not in favor of adding a dedicated GUI file browser because it would push Textadept's C core far beyond the 2000 line guideline. While earlier versions of Textadept did have a browser of sorts (one "mode" was for files; there were other modes), it was causing bloat issues and wasn't as extensible as I had originally hoped. I haven't missed it, but I do acknowledge the utility of such a feature.
